src/registrar/models/domain.py

```diff
@@ -667,6 +649,183 @@ def __str__(self) -> str:
+        streets = dict(
+            zip_longest(
+                ["street1", "street2", "street3"],
+                addr.street if addr is not None else [""],
+                fillvalue=None,
+            )
+        )
+
+        desired_contact = PublicContact(
+            domain=self,
+            contact_type=contact_type,
+            registry_id=contact_id,
+            email=contact.email or "",
+            voice=contact.voice or "",
+            fax=contact.fax,
+            name=postal_info.name or "",
+            org=postal_info.org,
+            # For linter - default to "" instead of None
+            pw=getattr(auth_info, "pw", ""),
+            city=getattr(addr, "city", ""),
+            pc=getattr(addr, "pc", ""),
+            cc=getattr(addr, "cc", ""),
+            sp=getattr(addr, "sp", ""),
+            **streets,
+        )  # type: ignore
+
+        return desired_contact
+
+    def _request_contact_info(self, contact: PublicContact):
+        try:
+            req = commands.InfoContact(id=contact.registry_id)
+            return registry.send(req, cleaned=True).res_data[0]
+        except RegistryError as error:
+            logger.error(
+                "Registry threw error for contact id %s contact type is %s, error code is\n %s full error is %s",  # noqa
+                contact.registry_id,
+                contact.contact_type,
+                error.code,
+                error,
+            )
+            raise error
+
+    def generic_contact_getter(
+        self, contact_type_choice: PublicContact.ContactTypeChoices
+    ) -> PublicContact | None:
+        """Abstracts the cache logic on EppLib contact items
+
+        contact_type_choice is a literal in PublicContact.ContactTypeChoices,
+        for instance: PublicContact.ContactTypeChoices.SECURITY.
+
+        If you wanted to setup getter logic for Security, you would call:
+        cache_contact_helper(PublicContact.ContactTypeChoices.SECURITY),
+        or cache_contact_helper("security").
+
+        Note: Registrant is handled slightly differently internally,
+        but the output will be the same.
+        """
+        # registrant_contact(s) are an edge case. They exist on
+        # the "registrant" property as opposed to contacts.
+        desired_property = "contacts"
+        if contact_type_choice == PublicContact.ContactTypeChoices.REGISTRANT:
+            desired_property = "registrant"
+
+        try:
+            contacts = self._get_property(desired_property)
+        except KeyError as error:
+            # Q: Should we be raising an error instead?
+            logger.error(f"Could not find {contact_type_choice}: {error}")
+            return None
+        else:
+            # Grab from cache
+            cached_contact = self.grab_contact_in_keys(contacts, contact_type_choice)
+            if cached_contact is None:
+                raise ValueError("No contact was found in cache or the registry")
```

Comment:
Currently the page just returns a 500. This actually brings up a good question regarding this stuff:
For our application, is it preferred to display as many errors as possible to the end user for really weird edge cases, or just display a 500? My logic here was that `if cached_contact is None:` should essentially never occur, and if it does it'd we'd need to: a) address it but also b) not display that to an end user who may use it maliciously.

However this may be unnecessary in this context, what is our general assumption with this kind of stuff?